Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/test_tools_fixtures.py

```diff
@@ -67,8 +67,6 @@ def _get_build_directory(config):
         logger.debug(f'Custom build directory set via cli arg to: {custom_build_directory}')
         if not os.path.exists(custom_build_directory):
             raise ValueError(f'Pytest custom argument "--build-directory" does not exist at: {custom_build_directory}')
-        if custom_build_directory.endswith('debug'):
-            pytest.exit("Python debug modules are not available. LyTestTools test skipped.", 0)
```

Comment:
Why was this change appropriate? To my knowledge the debug version of python continues to not function properly due to https://github.com/o3de/o3de/issues/4404

Reply:
See #9746.